unlock-app/src/components/interface/checkout/main/Confirm/ConfirmUniversalCard.tsx

```diff
@@ -0,0 +1,234 @@
+  const { isInitialLoading: isInitialDataLoading, data: purchaseData } =
+    usePurchaseData({
+      lockAddress: lock!.address,
+      network: lock!.network,
+      promo,
+      password,
+      captcha,
+      paywallConfig,
+      recipients,
+    })
+
+  // And now get the price to pay by card
+  const { data: cardPricing, isInitialLoading: isCardPricingLoading } =
+    useUniversalCardPrice({
+      network: lock!.network,
+      lockAddress: lock!.address,
+      recipients,
+      purchaseData: purchaseData!,
+      enabled: !isInitialDataLoading,
+    })
+
+  // TODO: Also configure webhook on the Stripe side?
+  const onChange = async (event: any) => {
+    const { session } = event
+
+    if (
+      session?.wallet_address &&
+      account &&
+      session?.wallet_address.toLowerCase() !== account.toLowerCase()
+    ) {
+      setSessionError(
+        'You cannot change the recipient address as it is the address that will receive the NFT membership token. Please start again.'
+      )
+    }
+
+    const expectedAmount = cardPricing?.total
+    if (
+      session.quote.destination_amount &&
+      100 * parseFloat(session.quote.destination_amount) !== expectedAmount
+    ) {
+      console.error(
+        'Price changed',
+        100 * parseFloat(session.quote.destination_amount),
+        expectedAmount
+      )
+      setSessionError('You cannot change the amount. Please start again.')
+    }
+
+    if (session.status === 'fulfillment_complete') {
+      const response = await storage.captureOnRampSession(
+        session.id,
+        session.quote.blockchain_tx_id
+      )
+      const transactionHash = response.data.transactionHash
+      if (transactionHash) {
+        onConfirmed(lock!.address, transactionHash)
+      }
+    }
+  }
+
+  // User triggers the payment!
+  const signPermit = async () => {
+    try {
+      const walletService = await getWalletService(lock!.network)
+      const {
+        transferSignature,
+        transferMessage,
+        purchaseSignature,
+        purchaseMessage,
+      } = await walletService.getAndSignAuthorizationsForTransferAndPurchase({
+        lockAddress: lock!.address,
+        network: lock!.network,
+        amount: cardPricing!.total.toString(), // amount is a string in cents
+      })
+
+      // We pass recipients and purchaseData as these will be used for the onchain transaction
+      const response = await storage.createOnRampSession(
+        lock!.network,
+        lock!.address,
+        {
+          transferSignature,
+          transferMessage,
+          purchaseSignature,
+          purchaseMessage,
+          recipients,
+          purchaseData: purchaseData!,
+        }
+      )
+      setOnrampSession(response.data)
+    } catch (error) {
+      console.error(error)
+      onError(`We could not initiate the card payment. Please try again!`)
+    }
+  }
+
+  if (isCardPricingLoading || !cardPricing) {
+    return null
```

Comment:
Do we want to add some loading spinner here? Anything would be fine for starters.

Reply:
yes